### ISSUE TITLE
kv::Value::to_$int should not use `as`

### ISSUE THREAD

**Issue by Thomasdezeeuw** [rust-lang/log#430]
This is the same issue as https://github.com/sval-rs/value-bag/issues/15, but I didn't get a response there so maybe I'll get one here since I'm not using value-bag directly.

The functions have two problems. First converting a larger integer to a smaller return the maximum value (in non-release mode), but it should return `None`, i.e. I expect the following code to work.

```rust
use value_bag::ValueBag;

fn main() {
    let value = ValueBag::from(u16::MAX);
    assert_eq!(None, value.to_u8());
}
```

But the above returns `Some(255)`. Second, it shouldn't convert signed to unsigned integers as that will also return the wrong result, i.e. the following.

```rust
use value_bag::ValueBag;

fn main() {
    let value = ValueBag::from(-123i64);
    assert_eq!(None, value.to_u64());
}
```

Returns `Some(18446744073709551493)`, again unexpected.

The same is true for `to_f64` and `to_f32`.

**Comment by KodrAus** [comment]
Responded over in https://github.com/sval-rs/value-bag/issues/15 🙇

**Comment by KodrAus** [comment]
This has been fixed as of `0.4.14`
